diag_manager/fms_diag_yaml.F90

```diff
@@ -1401,6 +1406,380 @@ subroutine dump_diag_yaml_obj( filename )
+    call fms_f2c_string(keys2(i)%key5, 'unlimdim')
+    call fms_f2c_string(keys2(i)%key6, 'new_file_freq')
+    call fms_f2c_string(keys2(i)%key7, 'new_file_freq_units')
+    !! gcc bug causes this key to sometimes not show up
+    call fms_f2c_string(keys2(i)%key8, 'start_time')
+    call fms_f2c_string(keys2(i)%key9, 'file_duration')
+    call fms_f2c_string(keys2(i)%key10, 'file_duration_units')
+    if (fileptr%has_file_fname())     call fms_f2c_string(vals2(i)%val1, fileptr%file_fname)
+    if (fileptr%has_file_unlimdim())  call fms_f2c_string(vals2(i)%val5, fileptr%file_unlimdim)
+    if (fileptr%has_file_timeunit()) then
+      call fms_f2c_string(vals2(i)%val4, get_diag_unit_string(fileptr%file_timeunit))
+    endif
+    if (fileptr%has_file_freq()) then
+      write(tmpstr1, '(I0)') fileptr%file_freq
+      call fms_f2c_string(vals2(i)%val2, trim(tmpstr1))
+    endif
+    if (fileptr%has_file_frequnit()) then
+      call fms_f2c_string(vals2(i)%val3, get_diag_unit_string(fileptr%file_frequnit))
+    endif
+    if(fileptr%has_file_new_file_freq()) then
+      write(tmpstr1, '(I0)') fileptr%file_new_file_freq
+      call fms_f2c_string(vals2(i)%val6, tmpstr1)
+    endif
+    if(fileptr%has_file_new_file_freq_units()) then
+      call fms_f2c_string(vals2(i)%val7, get_diag_unit_string(fileptr%file_new_file_freq_units))
+    endif
+    if(fileptr%has_file_start_time()) then
+      call fms_f2c_string(vals(i)%val8, trim(fileptr%get_file_start_time()))
+      st_vals(i) = fileptr%get_file_start_time()
+    endif
+    if(fileptr%has_file_duration()) then
+      write(tmpstr1, '(I0)') fileptr%file_duration
+      call fms_f2c_string(vals2(i)%val9, tmpstr1)
+    endif
+    if(fileptr%has_file_duration_units()) then
+      call fms_f2c_string(vals2(i)%val10, get_diag_unit_string(fileptr%file_duration_units))
+    endif
+    !! tier 3 - varlists, subregion, global metadata
+    call yaml_out_add_level2key('varlist', keys2(i))
+    j = 0
+    if( SIZE(fileptr%file_varlist) .gt. 0) then
+      do j=1, SIZE(fileptr%file_varlist)
+        key3_i = key3_i + 1
+        call initialize_key_struct(keys3(key3_i))
+        call initialize_val_struct(vals3(key3_i))
+        !! find the variable object from the list
+        varptr => NULL()
+        do varnum_i=1, SIZE(diag_yaml%diag_fields)
+          if(DEBUG) print *, 'diag_obj', trim(diag_yaml%diag_fields(varnum_i)%var_varname)
+          if(DEBUG) print *, 'file obj', trim(diag_yaml%diag_fields(varnum_i)%var_varname)
+          if( trim(diag_yaml%diag_fields(varnum_i)%var_varname ) .eq. trim(fileptr%file_varlist(j)) .and. &
+              trim(diag_yaml%diag_fields(varnum_i)%var_fname) .eq. trim(fileptr%file_fname)) then
+            if(DEBUG) print *, 'match'
+            varptr => diag_yaml%diag_fields(varnum_i)
+            exit
+          endif
+        enddo
+        if( .not. associated(varptr)) call mpp_error(FATAL, "diag_yaml_output: var name: "// &
```

Comment:
This is just printing a variable name.  A user running into this will have no idea what's gone wrong.